packages/xrpl/test/integration/transactions/oracleSet.test.ts

```diff
@@ -0,0 +1,78 @@
+import { stringToHex } from '@xrplf/isomorphic/dist/utils'
+import { assert } from 'chai'
+
+import { OracleSet } from '../../../src'
+import serverUrl from '../serverUrl'
+import {
+  setupClient,
+  teardownClient,
+  type XrplIntegrationTestContext,
+} from '../setup'
+import { testTransaction } from '../utils'
+
+// how long before each test case times out
+const TIMEOUT = 20000
+
+describe('OracleSet', function () {
+  let testContext: XrplIntegrationTestContext
+
+  beforeEach(async () => {
+    testContext = await setupClient(serverUrl)
+  })
+  afterEach(async () => teardownClient(testContext))
+
+  it(
+    'base',
+    async () => {
+      const tx: OracleSet = {
+        TransactionType: 'OracleSet',
+        Account: testContext.wallet.classicAddress,
+        OracleDocumentID: 1234,
+        LastUpdateTime: Math.floor(Date.now() / 1000),
+        PriceDataSeries: [
+          {
+            PriceData: {
+              BaseAsset: 'XRP',
+              QuoteAsset: 'USD',
+              AssetPrice: 740,
+              Scale: 3,
+            },
+          },
+        ],
+        Provider: stringToHex('chainlink'),
+        URI: '6469645F6578616D706C65',
+        AssetClass: stringToHex('currency'),
+      }
+
+      await testTransaction(testContext.client, tx, testContext.wallet)
+
+      // confirm that the Oracle was actually created
+      const getAggregatePriceResponse = await testContext.client.request({
```

Comment:
Why test with `get_aggregate_price` instead of `account_objects`? IMO the test would be a bit easier to read with just `account_objects`.